modules/libcom/src/error/errlog.c

```diff
@@ -466,17 +467,38 @@ int eltc(int yesno)
 {
     errlogInit(0);
     epicsMutexMustLock(pvt.msgQueueLock);
-    pvt.toConsole = yesno;
+    pvt.toConsole = yesno ? 1 : 0;
     epicsMutexUnlock(pvt.msgQueueLock);
     errlogFlush();
     return 0;
 }
 
 int errlogSetConsole(FILE *stream)
 {
+    int closeConsole = 0;
     errlogInit(0);
+    if (!stream) {
+        stream = stderr;
+    }
+#ifdef vxWorks
+    /* The stderr/stdout we get may become invalid as they are thread specific.
+     * Also, the FDs are actually maps to some "real" FDs.
+     * Thus, we clone the stream and link it with a dup of the "real" FD.
+     * The dup'ed FD will be closed when the FILE handle is fclosed. */
+    if (fileno(stream) <= 2) {
+        closeConsole = 1;
+        stream = fdopen(dup(ioTaskStdGet(0, fileno(stream))),"w");
+        if (!stream) {
+            return -1;
+        }
+    }
+#endif
     epicsMutexMustLock(pvt.msgQueueLock);
-    pvt.console = stream ? stream : stderr;
+    if (pvt.closeConsole) {
+        fclose(pvt.console);
+    }
```

Comment:
It bothers me slightly that this block has a conditional for a variable that's never going to be touched outside of vxworks; the "performance" isn't really relevant, but it still can make the code more confusing to readers. I wonder if it wouldn't make sense to add it under another `#ifdef` block, or something of the sort...

Reply:
I'm not bothered about an `#ifdef`.  However, since `fclose(NULL)` is not portable, paranoia would motivate me test `if(pvt.closeConsole && pvt.console)`.